### PR TITLE
[CHORE] Deep linking upates

### PR DESCRIPTION
**Pull request by evilpenguin**
Some minor updates from a PR after it was closed #149

### CODE REVIEW

File: SayTheirNames/Source/Dependency/Deep Linking/DeepLinkSupport.swift [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+
+import Foundation
+
+struct DeepLinkSupport {
```
Comment (hybridcattt) [inline SayTheirNames/Source/Dependency/Deep Linking/DeepLinkSupport.swift:27]:
```suggestion
enum DeepLinkSupport {
```

File: SayTheirNames/Source/Dependency/Deep Linking/Deep Links/AboutDeepLink.swift [hunk 2]
```diff
@@ -25,20 +25,13 @@
...
+    .path(DeepLinkSupport.Path.about)
     
-    init() {
-        self.linkDetails = type(of: self).details
-    }
-    
-    static func deepLink(fromComponents: [String]? = nil) -> DeepLink? {
-        return self.init()
-    }
+    init?(components: [String]?) { }
```
Comment (hybridcattt) [inline SayTheirNames/Source/Dependency/Deep Linking/Deep Links/AboutDeepLink.swift:36]: if this initializer never fails, it doesn't need to be failable (regular inits still satisfy the requirement)

File: SayTheirNames/Source/Dependency/Deep Linking/DeepLink.swift [hunk 3]
```diff
@@ -26,10 +26,9 @@ import UIKit
...
 protocol DeepLink {
     static var details: DeepLinkDetails { get }
-    static func deepLink(fromComponents: [String]?) -> DeepLink?
-    var linkDetails: DeepLinkDetails { get }
+    init?(components: [String]?)
 }
 
-protocol DeepLinkHandle {
-    func handle(deepLink: DeepLink)
+protocol DeepLinkPresentable {
```
Comment (hybridcattt) [inline SayTheirNames/Source/Dependency/Deep Linking/DeepLink.swift:32]: isn't it more like `DeepLinkPresenter`? "presentable" would be something that can be presented.

File: SayTheirNames/Source/Dependency/Deep Linking/DeepLinkDetails.swift [hunk 4]
```diff
@@ -55,35 +54,21 @@ public struct DeepLinkDetails {
...
-            else { return nil }
-            
-            for path in self.paths {
-                switch path {
-                case let .string(value, _):
-                    if value == location {
-                        components.remove(at: locationIndex)
-                        return self.type.deepLink(fromComponents: components)
-                    }
+    public func uses(scheme: String, host: String) -> Bool {
```
Comment (hybridcattt) [inline SayTheirNames/Source/Dependency/Deep Linking/DeepLinkDetails.swift:57]: It doesn't need public here - all in the same module

**Comment by mahmoudashraf93** [comment]
@hybridcattt had few comments after the PR was closed.

File: SayTheirNames/Source/Dependency/Deep Linking/DeepLinkHandler.swift [hunk 5]
```diff
@@ -33,14 +33,48 @@ final class DeepLinkHandler: Dependency {
...
+
+                // We are navigating to home
+                if components.count == 0 {
+                    return HomeDeepLink()
+                }
+                
+                // Check location
+                guard let location = components.first,
+                      let locationIndex = components.firstIndex(of: location)
+                else { return nil }
```
Comment (hybridcattt) [inline SayTheirNames/Source/Dependency/Deep Linking/DeepLinkHandler.swift:67]: Maybe it's just late, but I can't understand this check. it just checks the array is not empty and below removes the first element, no?

**Comment by evilpenguin** [comment]
> @hybridcattt had few comments after the PR was closed.

Indeed. These are in reference to those. I asked her to continue her review.

File: SayTheirNames/Source/Dependency/Deep Linking/DeepLinkSupport.swift (hunk 1 above)
Comment (evilpenguin) [reply SayTheirNames/Source/Dependency/Deep Linking/DeepLinkSupport.swift:27]: Done

File: SayTheirNames/Source/Dependency/Deep Linking/DeepLink.swift (hunk 3 above)
Comment (evilpenguin) [reply SayTheirNames/Source/Dependency/Deep Linking/DeepLink.swift:32]: Done.

File: SayTheirNames/Source/Dependency/Deep Linking/DeepLinkDetails.swift (hunk 4 above)
Comment (evilpenguin) [reply SayTheirNames/Source/Dependency/Deep Linking/DeepLinkDetails.swift:57]: Habit to type public.

Comment (evilpenguin) [reply SayTheirNames/Source/Dependency/Deep Linking/DeepLinkDetails.swift:57]: Done

File: SayTheirNames/Source/Dependency/Deep Linking/DeepLinkHandler.swift (hunk 5 above)
Comment (evilpenguin) [reply SayTheirNames/Source/Dependency/Deep Linking/DeepLinkHandler.swift:67]: It pulls the location to check for the deep link type later, and then removes it.

File: SayTheirNames/Source/Dependency/Deep Linking/Deep Links/AboutDeepLink.swift (hunk 2 above)
Comment (evilpenguin) [reply SayTheirNames/Source/Dependency/Deep Linking/Deep Links/AboutDeepLink.swift:36]: Done

**Pull request merged**